app/networkapi/milestones/models.py

```diff
@@ -0,0 +1,74 @@
+from django.utils import timezone
+from django.db import models
+
+from networkapi.utility.images import get_image_upload_path
+
+
+def get_milestone_photo_upload_path(instance, filename):
+    return get_image_upload_path(
+        app_name='milestones',
+        prop_name='headline',
+        instance=instance,
+        current_filename=filename
+    )
+
+
+class MilestoneLink(models.Model):
+    label = models.CharField(
+        max_length=300,
+        help_text='Text to describe the milestone link'
+    )
+    url = models.URLField(
+        max_length=500,
+        help_text='URL link to the milestone',
+    )
+
+    def __str__(self):
+        return str(self.label)
+
+
+class MilestoneQuerySet(models.query.QuerySet):
+    """
+    A QuerySet that filters for upcoming milestones
+    """
+    def upcoming(self):
+        now = timezone.now()
+        return self.filter(
+            end_date__gte=now,
+        )
+
+
+class Milestone(models.Model):
+    headline = models.CharField(
+        max_length=300,
+        help_text='Title of the milestone'
+    )
+    photo = models.FileField(
+        max_length=2048,
+        help_text='Image associated with the milestone',
+        upload_to=get_milestone_photo_upload_path,
+    )
+    start_date = models.DateField(
+        help_text='Date on which the milestone begins'
+    )
+    end_date = models.DateField(
+        help_text='Date on which the milestone ends'
+    )
+    description = models.TextField(
+        help_text='Briefly describe what the milestone '
+                  'is all about',
+    )
+    link = models.OneToOneField(
+        'MilestoneLink',
+        null=True,
+        on_delete=models.CASCADE
+    )
+
```

Comment:
An alternative to this is to use link_url and link_label directly as property. We need a null = True beacuase one of them (milestone, milestonelink)is needed to be save first, which throws an error as corresponding relation dont exist